packages/wrangler/src/index.tsx

```diff
@@ -523,10 +523,16 @@ function createCLIParser(argv: string[]) {
 				(args.config as ConfigPath) ||
 				(args.script && findWranglerToml(path.dirname(args.script)));
 			const config = readConfig(configPath, args);
-			await metrics.sendMetricsEvent("deploy worker script", {
-				sendMetrics: config.send_metrics,
-			});
 			const entry = await getEntry(args, config, "publish");
+			await metrics.sendMetricsEvent(
+				"deploy worker script",
+				{
+					usesTypeScript: /\.tsx?$/.test(entry.file),
```

Comment:
This won't work if the entry is a js file but it imports a ts file

Reply:
This is true. How common is this scenario?

I looked into using the inputs in the esbuild `metafile` but this is problematic. First we have to thread this information around the codebase a bit. Second, this bundling step is async and runs within the lifetime of the InkJS app, which can cause a dialog box to pop up in the middle of the UI if the user has not yet opted-in/out of metrics sending.

I think that the main entry point being a TS file is a strong enough signal for what we are trying to analyse in terms of usage.